OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/ResetPasswordAlertDialogFragment.java

```
@@ -0,0 +1,200 @@
+        til = new TextInputLayout(getContext());
+        til.addView(et);
+        til.setHint(getContext().getString(R.string.email));
+
+        int paddingWidth = Math.round(getResources().getDimension(R.dimen.alert_dialog_padding_width));
+        int paddingHeight = Math.round(getResources().getDimension(R.dimen.alert_dialog_padding_height));
+        til.setPadding(paddingWidth, paddingHeight, paddingWidth, paddingHeight);
+
+        alertDialog.setView(til);
+
+        String email = getArguments().getString(EXTRA_LOGIN_EMAIL);
+        if (email != null && !email.isEmpty()) {
+            et.setText(email);
+        }
+        return alertDialog;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        /**
+         * This is a way to prevent the alert dialog from auto-dismissing when the positive/negative button is pressed
+         * We find the positive button (must be after it is created), then manually override the onClickListener
+         */
+        ((AlertDialog) getDialog()).getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                submit(getEnteredEmail());
+            }
+        });
+    }
+
+
+    @Override
+    public void onDismiss(DialogInterface dialog) {
+        super.onDismiss(dialog);
+        if (listener!=null) {
+            listener.omDismissed();
+        }
+    }
+
+    public void setListener(@Nullable Listener listener) {
+        this.listener = listener;
+    }
+
+    @Nullable
+    protected String getEnteredEmail() {
+        String email = null;
+
+        if (et != null) {
+            email = et.getText().toString().trim();
+        }
+
+        return email;
+    }
+
+    /**
+     * Shows or hides error
+     * @param error The error to show. If null, error will be hidden
+     */
+    public void showError(@Nullable String error) {
```

According to the doc string `@param error The error to show. If null, error will be hidden`, showError seems to be a misnomer if it is also being used to dismiss errors.